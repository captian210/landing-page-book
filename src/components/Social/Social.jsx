import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {
	FacebookShareButton,
	FacebookIcon,
	GooglePlusShareButton,
	GooglePlusIcon,
	LinkedinShareButton,
	LinkedinIcon,
	TwitterShareButton,
	TwitterIcon,
	TelegramShareButton,
	TelegramIcon,
	WhatsappShareButton,
	WhatsappIcon,
	PinterestShareButton,
	PinterestIcon,
	VKShareButton,
	VKIcon,
	OKShareButton,
	OKIcon,
	RedditShareButton,
	RedditIcon,
	TumblrShareButton,
	TumblrIcon,
	LivejournalShareButton,
	LivejournalIcon,
	ViberShareButton,
	ViberIcon,
	EmailShareButton,
	EmailIcon
} from 'react-share';

class Social extends Component {

	render() {
		const {
			url,
			rounded,
			facebook,
			google,
			linkedin,
			twitter,
			telegram,
			whatsapp,
			pinterest,
			vk,
			ok,
			reddit,
			tumblr,
			livejournal,
			viber,
			email,
		} = this.props;
		return <div className="Social">
			{
				facebook &&
				<FacebookShareButton url={url}>
					<FacebookIcon round={rounded}/>
				</FacebookShareButton>
			}
			{
				google &&
				<GooglePlusShareButton url={url}>
					<GooglePlusIcon round={rounded}/>
				</GooglePlusShareButton>
			}
			{
				linkedin &&
				<LinkedinShareButton url={url}>
					<LinkedinIcon round={rounded}/>
				</LinkedinShareButton>
			}
			{
				twitter &&
				<TwitterShareButton url={url}>
					<TwitterIcon round={rounded}/>
				</TwitterShareButton>
			}
			{
				telegram &&
				<TelegramShareButton url={url}>
					<TelegramIcon round={rounded}/>
				</TelegramShareButton>
			}
			{
				whatsapp &&
				<WhatsappShareButton url={url}>
					<WhatsappIcon round={rounded}/>
				</WhatsappShareButton>
			}
			{
				pinterest &&
				<PinterestShareButton url={url}>
					<PinterestIcon round={rounded}/>
				</PinterestShareButton>
			}
			{
				vk &&
				<VKShareButton url={url}>
					<VKIcon round={rounded}/>
				</VKShareButton>
			}
			{
				ok &&
				<OKShareButton url={url}>
					<OKIcon round={rounded}/>
				</OKShareButton>
			}
			{
				reddit &&
				<RedditShareButton url={url}>
					<RedditIcon round={rounded}/>
				</RedditShareButton>
			}
			{
				tumblr &&
				<TumblrShareButton url={url}>
					<TumblrIcon round={rounded}/>
				</TumblrShareButton>
			}
			{
				livejournal &&
				<LivejournalShareButton url={url}>
					<LivejournalIcon round={rounded}/>
				</LivejournalShareButton>
			}
			{
				viber &&
				<ViberShareButton url={url}>
					<ViberIcon round={rounded}/>
				</ViberShareButton>
			}
			{
				email &&
				<EmailShareButton url={url}>
					<EmailIcon round={rounded}/>
				</EmailShareButton>
			}
		</div>
	}
}

Social.propTypes = {
	facebook: PropTypes.bool,
	google: PropTypes.bool,
	linkedin: PropTypes.bool,
	twitter: PropTypes.bool,
	telegram: PropTypes.bool,
	whatsapp: PropTypes.bool,
	pinterest: PropTypes.bool,
	vk: PropTypes.bool,
	ok: PropTypes.bool,
	reddit: PropTypes.bool,
	tumblr: PropTypes.bool,
	livejournal: PropTypes.bool,
	viber: PropTypes.bool,
	email: PropTypes.bool,
	rounded: PropTypes.bool,
	url: PropTypes.string,
};
Social.defaultProps = {
	facebook: true,
	google: true,
	linkedin: true,
	twitter: true,
	telegram: true,
	whatsapp: true,
	pinterest: true,
	vk: true,
	ok: true,
	reddit: true,
	tumblr: true,
	livejournal: true,
	viber: true,
	email: true,
	rounded: true,
	url: ""
};

export default Social;
