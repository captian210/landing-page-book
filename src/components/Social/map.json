{"description":"","displayName":"Social","methods":[],"props":{"gray":{"type":{"name":"bool"},"required":false,"description":"","defaultValue":{"value":"false","computed":false}},"rounded":{"type":{"name":"bool"},"required":false,"description":"","defaultValue":{"value":"true","computed":false}},"type":{"type":{"name":"string"},"required":false,"description":"","defaultValue":{"value":"'email'","computed":false}},"url":{"type":{"name":"string"},"required":false,"description":"","defaultValue":{"value":"''","computed":false}},"directLink":{"type":{"name":"bool"},"required":false,"description":"","defaultValue":{"value":"true","computed":false}}},"filename":"/components/Social/Social.jsx"}