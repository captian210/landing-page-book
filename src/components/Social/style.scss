.Social{
  .SocialMediaShareButton{
    display: inline-block;
  }
}