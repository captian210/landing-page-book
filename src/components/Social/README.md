# Social

[![standard-readme compliant](https://img.shields.io/badge/standard--readme-OK-green.svg?style=flat-square)](https://github.com/RichardLitt/standard-readme)

## Overview
`@front10/landing-page-book/Social` is used to share in social networks.

## How to use
1- Importing bootstrap style

```js
import "bootstrap/dist/css/bootstrap.min.css";
```
2- Add following content to import the component:

```js
import Social from "@front10/landing-page-book/dist/components/Social";
import "@front10/landing-page-book/dist/components/Social/style.css";
```

> Note: For including all components styles once you can use [a theme](https://github.com/front10/landing-page-book/wiki/Theming).

3- Put this code into jsx page:
```html
<Social url="https://front10.com"
        type="facebook"/>
```

[![Edit Social demo](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/zrrz8wr81l)

## Properties:

| </br>Name   | </br>Type | </br>Summary                                                                                 | 
| ------------| - | ------------------------------------------------------------------------------------------------------ |
| gray      | `Boolean` | Define if image is gray scale. Default `false` |
| rounded      | `Boolean` | Show buttons rounded or square. Default `true` |
| type      | `String` | Type of button. Default `"email"`, possibles values are: `facebook`, `google`, `linkedin`, `twitter`, `telegram`, `whatsapp`, `pinterest`, `vk`, `ok`, `reddit`, `tumblr`, `livejournal`, `email`, `viber` |
| url      | `String` | Url to share. Default `""` |
| directLink      | `Boolean` | Indicates if it is redirected to a link within the social network or a floating window is displayed to share the link using the social network. Default `true` |
