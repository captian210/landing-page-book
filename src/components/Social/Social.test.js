import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Social from './Social';

configure({ adapter: new Adapter() });

describe('Social', () => {
  let props;
  let mounted;
  const social = () => {
    if (!mounted) {
      mounted = shallow(<Social {...props} />);
    }
    return mounted;
  };
  describe('Social', () => {
    beforeEach(() => {
      props = {};
      mounted = undefined;
    });

    it('should render', () => {
      expect(shallow(<Social {...props} />)).toMatchSnapshot();
    });
  });
});
