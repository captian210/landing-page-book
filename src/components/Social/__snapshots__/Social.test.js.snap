// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social Social should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Social
    gray={false}
    rounded={true}
    type="email"
    url=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        <CreatedButton
          className=""
          onClick={[Function]}
          openWindow={false}
          url=""
        >
          <Icon
            className=""
            iconBgStyle={Object {}}
            logoFillColor="white"
            round={true}
            size={64}
          />
        </CreatedButton>,
      ],
    },
    "ref": null,
    "rendered": Array [
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      false,
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Icon
            className=""
            iconBgStyle={Object {}}
            logoFillColor="white"
            round={true}
            size={64}
          />,
          "className": "",
          "onClick": [Function],
          "openWindow": false,
          "url": "",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "",
            "iconBgStyle": Object {},
            "logoFillColor": "white",
            "round": true,
            "size": 64,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          false,
          <CreatedButton
            className=""
            onClick={[Function]}
            openWindow={false}
            url=""
          >
            <Icon
              className=""
              iconBgStyle={Object {}}
              logoFillColor="white"
              round={true}
              size={64}
            />
          </CreatedButton>,
        ],
      },
      "ref": null,
      "rendered": Array [
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        false,
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Icon
              className=""
              iconBgStyle={Object {}}
              logoFillColor="white"
              round={true}
              size={64}
            />,
            "className": "",
            "onClick": [Function],
            "openWindow": false,
            "url": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "iconBgStyle": Object {},
              "logoFillColor": "white",
              "round": true,
              "size": 64,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
